source/client-side-encryption/client-side-encryption.rst

```diff
@@ -591,6 +623,70 @@ the user. Refer:
+   It protects against a malicious server advertising a false ``encryptedFields``.
+
+.. _fle2-createcollection-drop:
+
+FLE 2 ``CreateCollection()`` and ``Collection.Drop()``
+------------------------------------------------------
+
+A collection supporting FLE 2 requires an index and three additional collections.
+
+Drivers MUST have a BSON document option named ``encryptedFields`` in ``CreateCollection()``.
+
+A call to a driver helper ``CreateCollection(collectionName, collectionOptions)`` must check if the collection namespace (``<databaseName>.<collectionName>``) has an associated ``encryptedFields``. Check for an associated ``encryptedFields`` from the following:
+
+- The ``encryptedFields`` option passed in ``collectionOptions``.
+- The value of ``AutoEncryptionOpts.encryptedFieldsMap[<databaseName>.<collectionName>]``.
+
+If the collection namespace has an associated ``encryptedFields``, then do the following operations. If any of the following operations error, the remaining operations are not attempted:
+
+- Create the collection with name ``encryptedFields["escCollection"]`` using default options.
+  If ``encryptedFields["escCollection"]`` is not set, use the collection name ``enxcol_.<collectionName>.esc``.
+  Creating this collection MUST NOT check if the collection namespace is in the ``AutoEncryptionOpts.encryptedFieldsMap``.
+- Create the collection with name ``encryptedFields["eccCollection"]`` using default options.
+  If ``encryptedFields["eccCollection"]`` is not set, use the collection name ``enxcol_.<collectionName>.ecc``.
+  Creating this collection MUST NOT check if the collection namespace is in the ``AutoEncryptionOpts.encryptedFieldsMap``.
+- Create the collection with name ``encryptedFields["ecocCollection"]`` using default options.
+  If ``encryptedFields["ecocCollection"]`` is not set, use the collection name ``enxcol_.<collectionName>.ecoc``.
+  Creating this collection MUST NOT check if the collection namespace is in the ``AutoEncryptionOpts.encryptedFieldsMap``.
+- Create the collection ``collectionName`` with ``collectionOptions`` and the option ``encryptedFields`` set to the ``encryptedFields``.
+- Create the the index ``{"__safeContent__": 1}`` on collection ``collectionName``.
+
+Drivers MUST have a BSON document option named ``encryptedFields`` in ``Collection.Drop()``.
+
+A call to a driver helper ``Collection.Drop(dropOptions)`` must check if the collection namespace (``<databaseName>.<collectionName>``) has an associated ``encryptedFields``. Check for an associated ``encryptedFields`` from the following:
+
+- The ``encryptedFields`` option passed in ``dropOptions``.
+- The value of ``AutoEncryptionOpts.encryptedFieldsMap[<databaseName>.<collectionName>]``.
+- If ``AutoEncryptionOpts.encryptedFieldsMap`` is not null, run a ``listCollections`` command on the database ``databaseName`` with the filter ``{ "name": "<collectionName>" }``. Check the returned ``options`` for the ``encryptedFields`` option.
```

Comment:
My gut feeling here would be:
- Dropping collections is a fairly infrequent operation
- `listCollections` with a filter for a single collection is comparatively cheap
- The downsides of leaving unused collections lying around are larger than the upsides of fast collection dropping

I know it adds an extra round-trip, but I’d be okay with dropping the base condition here. (If not, that’s also fine – we’ll implement this behavior in devtools either way.)

Reply:
SGTM – Feel free to leave open if you’d like more input here, otherwise you can just resolve this comment